catroid/src/main/java/org/catrobat/catroid/content/bricks/WhenGamepadButtonBrick.java

```diff
@@ -0,0 +1,163 @@
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.catrobat.catroid.content.bricks;
+
+import android.content.Context;
+import android.view.View;
+import android.widget.AdapterView;
+import android.widget.ArrayAdapter;
+import android.widget.BaseAdapter;
+import android.widget.CompoundButton;
+import android.widget.Spinner;
+
+import com.badlogic.gdx.scenes.scene2d.actions.SequenceAction;
+
+import org.catrobat.catroid.CatroidApplication;
+import org.catrobat.catroid.R;
+import org.catrobat.catroid.content.Script;
+import org.catrobat.catroid.content.Sprite;
+import org.catrobat.catroid.content.WhenGamepadButtonScript;
+
+import java.util.List;
+
+public class WhenGamepadButtonBrick extends BrickBaseType implements ScriptBrick {
+
+	protected WhenGamepadButtonScript whenGamepadButtonScript;
+	protected transient boolean checked = false;
+	private static final long serialVersionUID = 1L;
+	private String[] actions;
+	private String action;
+	private int position;
+
+	public WhenGamepadButtonBrick(WhenGamepadButtonScript whenGamepadButtonScript) {
+		this.whenGamepadButtonScript = whenGamepadButtonScript;
+		actions = CatroidApplication.getAppContext().getResources().getStringArray(R.array.gamepad_buttons_array);
+		if (whenGamepadButtonScript != null) {
+			this.action = whenGamepadButtonScript.getAction();
+			for (int i = 0; i < actions.length; i++) {
+				if (actions[i].equals(this.action)) {
+					this.position = i;
+					break;
+				}
+			}
+		}
+	}
+
+	@Override
+	public int getRequiredResources() {
+		return CAST_REQUIRED;
+	}
+
+	@Override
+	public Brick copyBrickForSprite(Sprite sprite) {
+		WhenGamepadButtonBrick copyBrick = (WhenGamepadButtonBrick) clone();
+		copyBrick.whenGamepadButtonScript = whenGamepadButtonScript;
+		return copyBrick;
+	}
+
+	@Override
+	public View getView(final Context context, int brickId, final BaseAdapter baseAdapter) {
+		if (animationState) {
+			return view;
+		}
+
+		view = View.inflate(context, R.layout.brick_when_gamepad_button, null);
+
+		final Brick brickInstance = this;
+		setCheckboxView(R.id.brick_when_gamepad_button_checkbox);
+		checkbox.setOnCheckedChangeListener(new CompoundButton.OnCheckedChangeListener() {
+
+			@Override
+			public void onCheckedChanged(CompoundButton buttonView, boolean isChecked) {
+				checked = isChecked;
+				adapter.handleCheck(brickInstance, isChecked);
+			}
+		});
+
+		final Spinner actionSpinner = (Spinner) view.findViewById(R.id.brick_when_gamepad_button_spinner);
+		actionSpinner.setFocusableInTouchMode(false);
+		actionSpinner.setFocusable(false);
+		if (!(checkbox.getVisibility() == View.VISIBLE)) {
```

Comment:
isClickable is set true anyways in line 113, so can this be removed from the if?